validator/client/validator.go

```diff
@@ -229,7 +229,7 @@ func (v *validator) RolesAt(slot uint64) map[[48]byte]pb.ValidatorRole {
 		if assignment == nil {
 			role = pb.ValidatorRole_UNKNOWN
 		}
-		if assignment.Slot == slot {
+		if assignment.AttesterSlot == slot {
 			// Note: A proposer also attests to the slot.
 			if assignment.IsProposer {
```

Comment:
This logic is no longer correct since proposer slot may not equal attester slot. 

1) a pub key may have two roles at one slot
2) a pub key may be an attester at slot, but not a proposer and the inverse is true as well

Reply:
Discussed offline but the inverse is not true. A validator with an assignment is always an attester and optionally a proposer.